server/lib/pbench/__init__.py

```diff
@@ -141,6 +141,18 @@ def format(self, record):
         return msg
 
 
+class PbenchFormatter:
+    """Used for importing internal Pbench formatter and StyleAdapter
+    for test logger
+    """
+
+    def format(self, log_fmt: str, handler: Handler, logger: Logger) -> Logger:
+        formatter = _PbenchLogFormatter(fmt=log_fmt)
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+        return _StyleAdapter(logger)
+
+
```

Comment:
I added this here because I didn't want internal modules to be called and imported somewhere else. If I add it in test-specific module than the problem will be same, I will have to import these internal modules and call them.

Reply:
I added this here because I didn't want internal modules to be called and imported somewhere else. If I add it in test-specific module than the problem will be same, I will have to import these internal modules and call them.

Reply:
That's excellent reasoning.

So, we're left with a set of conflicting constraints:  we shouldn't violate module "boundaries" but we also should do our best to avoid adding testing support to production code.

I support your line of thinking in general (in production code, in particular), but unit testing is special.  Unit testing is intended to target a particular function, which could be an internal interface, and so it is bounded only by language and runtime constraints.  (That is, unit testing is not above a certain level of hackery -- e.g., mocks! -- and so it is OK for it to violate the underscore prefix convention.)  You can think of unit test code as having "internal" access, even though it happens to be packaged in a separate module.  (Lots of projects actually include the unit test functions inside the module that they test, for just such reason...although, I prefer maintaining them separately.)

So, really, it's perfectly acceptable for unit test code to reference "private" functions and classes.  And, doing so is better (IMO) than adding test code to production modules.

Reply:
That sounds perfect!